src/rounded_rect.rs

```diff
@@ -0,0 +1,302 @@
+            ),
+            build_arc_iter(
+                0,
+                Point {
+                    x: self.rect.x1 - radius,
+                    y: self.rect.y1 - radius,
+                },
+            ),
+            build_arc_iter(
+                1,
+                Point {
+                    x: self.rect.x0 + radius,
+                    y: self.rect.y1 - radius,
+                },
+            ),
+        ];
+
+        let rect = RectPathIter {
+            rect: self.rect,
+            ix: 0,
+            radius: self.radius,
+        };
+
+        RoundedRectPathIter { idx: 0, rect, arcs }
+    }
+
+    #[inline]
+    fn area(&self) -> f64 {
+        let radius = self.radius();
+        self.rect.area() - (4.0 - PI) * radius * radius
+    }
+
+    #[inline]
+    fn perimeter(&self, _accuracy: f64) -> f64 {
+        let radius = self.radius();
+        2.0 * (self.width().abs() + self.height().abs()) - 8.0 * radius + 2.0 * PI * radius
+    }
+
+    #[inline]
+    fn winding(&self, pt: Point) -> i32 {
+        let radius = self.radius();
+        let inside_half_width = (self.width() / 2.0 - self.radius()).max(0.0);
+        let inside_half_height = (self.height() / 2.0 - self.radius()).max(0.0);
+
+        // project point out of the inner rectangle (positive quadrant)
+        let px = (pt.x.abs() - inside_half_width).max(0.0);
+        let py = (pt.y.abs() - inside_half_height).max(0.0);
+
+        let inside = px * px + py * py <= radius * radius;
+        if inside {
+            1
+        } else {
+            0
+        }
+    }
+
+    #[inline]
+    fn bounding_box(&self) -> Rect {
+        self.rect.bounding_box()
+    }
+
+    #[inline]
+    fn as_rounded_rect(&self) -> Option<RoundedRect> {
+        Some(*self)
+    }
+}
+
+struct RectPathIter {
+    rect: Rect,
+    radius: f64,
+    ix: usize,
+}
+
+// This is clockwise in a y-down coordinate system for positive area.
+impl Iterator for RectPathIter {
+    type Item = PathEl;
+
+    fn next(&mut self) -> Option<PathEl> {
+        self.ix += 1;
+        match self.ix {
+            1 => Some(PathEl::MoveTo(Point::new(
+                self.rect.x0,
+                self.rect.y0 + self.radius,
+            ))),
+            2 => Some(PathEl::LineTo(Point::new(
+                self.rect.x1 - self.radius,
+                self.rect.y0,
+            ))),
+            3 => Some(PathEl::LineTo(Point::new(
+                self.rect.x1,
+                self.rect.y1 - self.radius,
+            ))),
+            4 => Some(PathEl::LineTo(Point::new(
+                self.rect.x0 + self.radius,
+                self.rect.y1,
+            ))),
+            5 => Some(PathEl::ClosePath),
+            _ => None,
+        }
+    }
+}
+
+// This is clockwise in a y-down coordinate system for positive area.
+impl Iterator for RoundedRectPathIter {
+    type Item = PathEl;
+
+    fn next(&mut self) -> Option<PathEl> {
+        if self.idx > 4 {
+            return None;
```

Comment:
Does this miss the `ClosePath` at the end? I'm finding it a bit hard to follow the logic.

Reply:
Yes, also verified it visually. Added a few more inline comments for better explanation.